src/common/unicode_util.cpp

```diff
@@ -1711,7 +1711,16 @@ UnicodeUtil::ICU* UnicodeUtil::Utf16Collation::loadICU(
 			}
 
 			if (avail < 0)
-				continue;
+			{
+				UErrorCode status = U_ZERO_ERROR;
+				UCollator* testCollator = icu->ucolOpen(locale.c_str(), &status);
+				if (!testCollator)
+					continue;
+
+				icu->ucolClose(testCollator);
+				if (U_FAILURE(status) || (status == U_USING_DEFAULT_WARNING))
```

Comment:
I tried `create collation unicode_ptbr for utf8 from unicode 'LOCALE=pt_BRx'` and it worked, but should not.

Can you change that to `if (status != U_ZERO_ERROR)` to treat warnings as errors?

Reply:
Oops, good catch. Thank you!

While I had tested with completely invalid locale names, I missed testing with partially valid ones. Changed as requested; now your example also fails (and the examples from my use cases continue to work).